mypy/server/deps.py

```diff
@@ -264,6 +283,18 @@ def visit_assignment_stmt(self, o: AssignmentStmt) -> None:
             assert info.typeddict_type is not None
             prefix = '%s.%s' % (self.scope.current_full_target(), info.name())
             self.add_type_dependencies(info.typeddict_type, target=make_trigger(prefix))
+        elif isinstance(rvalue, IndexExpr) and isinstance(rvalue.analyzed, TypeAliasExpr):
+            assert len(o.lvalues) == 1
+            lvalue = o.lvalues[0]
+            assert isinstance(lvalue, NameExpr)
+            attr_trigger = make_trigger('%s.%s' % (self.scope.current_full_target(),
```

Comment:
It would be better to avoid constructing triggers by concatenating strings, since there are edge cases where this is not correct (such as for things nested within functions). It's better to use the full name instead. Even if this is not always correct yet, at least we can hopefully fix it in one place (where we initialize the full name).